src/main/java/bio/terra/janitor/app/controller/UnauthenticatedApiController.java

```diff
@@ -17,27 +18,48 @@
 @Controller
 public class UnauthenticatedApiController implements UnauthenticatedApi {
   private final NamedParameterJdbcTemplate jdbcTemplate;
+  private final StairwayComponent stairwayComponent;
 
   @Autowired
-  UnauthenticatedApiController(JanitorJdbcConfiguration jdbcConfiguration) {
+  UnauthenticatedApiController(
+      JanitorJdbcConfiguration jdbcConfiguration, StairwayComponent stairwayComponent) {
     this.jdbcTemplate = new NamedParameterJdbcTemplate(jdbcConfiguration.getDataSource());
+    this.stairwayComponent = stairwayComponent;
   }
 
   @Override
   public ResponseEntity<SystemStatus> serviceStatus() {
-    if (jdbcTemplate.getJdbcTemplate().execute((Connection connection) -> connection.isValid(0))) {
-      return new ResponseEntity<>(
-          new SystemStatus()
-              .ok(true)
-              .putSystemsItem("postgres", new SystemStatusSystems().ok(true)),
-          HttpStatus.OK);
+    SystemStatus systemStatus = new SystemStatus();
+
+    final boolean postgresOk =
+        jdbcTemplate.getJdbcTemplate().execute((Connection connection) -> connection.isValid(0));
+    systemStatus.putSystemsItem("postgres", new SystemStatusSystems().ok(postgresOk));
+
+    StairwayComponent.Status stairwayStatus = stairwayComponent.getStatus();
+    final boolean stairwayOk = stairwayStatus.equals(StairwayComponent.Status.OK);
+    systemStatus.putSystemsItem(
+        "stairway",
+        new SystemStatusSystems().ok(stairwayOk).addMessagesItem(stairwayStatus.toString()));
+
+    systemStatus.ok(postgresOk && stairwayOk);
+    if (systemStatus.isOk()) {
+      return new ResponseEntity<>(systemStatus, HttpStatus.OK);
     } else {
-      return new ResponseEntity<>(
-          new SystemStatus()
-              .ok(false)
-              .putSystemsItem("postgres", new SystemStatusSystems().ok(false)),
-          HttpStatus.INTERNAL_SERVER_ERROR);
+      return new ResponseEntity<>(systemStatus, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  @Override
+  public ResponseEntity<Void> shutdownRequest() {
+    try {
+      if (!stairwayComponent.shutdown()) {
```

Comment:
are we shutting down everything or just stairway connection? That sounds this endpoint is partially done yet? Or this is just the endpoint  should be called "shutDownStariway"?

Reply:
The idea is to have k8s call this endpoint when it's going to terminate the pod (when it's being nice, there are no guarantees). This lets us leave around less garbage to recover later. Stairway has support for that, so we're calling Stairway. We might also stop scheduling our chronic tasks once we implement them.
Tried to add some comments, let me know if more would be helpful.